### ISSUE TITLE
OffsetPaging. The compilation is unable to resolve cyclic dependencies

### ISSUE THREAD

**Issue by ademchenko** [ChilliCream/hotchocolate#2883]
We are in the process of moving our preproduction services to v11. And we're facing a really big issue which is a blocker for us.

Suppose, we have a collection of Users but each user has its own collection of users, say 'Parents'. It's enough to get the compilation failed. 
@michaelstaib this is a really critical issue so I kindly ask you to give enough priority to that issue to fix it ASAP. I've made a complete test reproducing that error.

```
public class Issue2883Reproducing
    {
        private readonly ITestOutputHelper _testOutputHelper;

        public class User
        {
            public string? FirstName { get; set; }

            public List<User> Parents { get; set; }
        }

        public class UserType : ObjectType<User>
        {
            protected override void Configure(IObjectTypeDescriptor<User> descriptor)
            {
                descriptor.Field(i => i.Parents).UseOffsetPaging<UserType>().Resolve((_1, _2) => new[] { new User { FirstName = "Mother" }, new User { FirstName = "Father" } });
            }
        }

        public class Query
        {
            public List<User> Users => new()
            {
                new User()
            };
        }

        public class QueryType : ObjectType<Query>
        {
            protected override void Configure(IObjectTypeDescriptor<Query> descriptor)
            {
                descriptor.Field(t => t.Users)
                    .UseOffsetPaging<UserType>();
            }
        }

        public Issue2883Reproducing(ITestOutputHelper testOutputHelper)
        {
            _testOutputHelper = testOutputHelper;
        }

        [Fact]
        public async Task Test()
        {
            IRequestExecutor executor =
                await new ServiceCollection()
                    .AddGraphQL()
                    .AddQueryType<QueryType>()
                    .SetPagingOptions(new PagingOptions { DefaultPageSize = 50 })
                    .Services
                    .BuildServiceProvider()
                    .GetRequestExecutorAsync();

            IExecutionResult executionResult = await executor
                .ExecuteAsync(@"
                {
                    users {
                        items
                        {
                            parents
                            {
                                items
                                {
                                    firstName
                                }
                            }
                       }
                    }
                }");

            foreach (IError error in executionResult.Errors ?? new IError[0])
                _testOutputHelper.WriteLine(error.Message);

            Assert.Equal(0, executionResult.Errors?.Count ?? 0);
        }
    }
```

**Comment by michaelstaib** [comment]
This one is on our list and will be fixed with 11.0.9 probably end of next week

**Comment by ademchenko** [comment]
@michaelstaib, any updates on that blocker issue?

**Comment by ademchenko** [comment]
@michaelstaib , I have downloaded version 11.0.9 and run the test above. Nothing has changed. The version remains absolutely broken and unusable. It has taken 2 weeks already and that critical bug still exists leaving us in the middle of moving to v.11 process.
Do you have any real estimates about fixing that?

**Comment by michaelstaib** [comment]
Hi @ademchenko we have moved the fix to 11.0.10.

I cannot give you an exact date as we are at the moment fully focused on strawberry shake. But a rough timeframe for 11.0.10 is in the next 14 days.

**Comment by ademchenko** [comment]
Hi, @michaelstaib, thanks, it works in v11.1.0-preview.59. Hope it will be brought into the production shortly.
